### DISCUSSION TITLE
How to keep the file info with tar.gz?

### DISCUSSION THREAD

**Discussion by nick-delirium** [tsolomko/SWCompression#47] [Q&A] [answered]
What I'm trying to achieve is to create an archive from a list of jpeg image data, then send it to backend and process further, but for testing I'm actually saving it locally for now. I noticed that after unarchiving, all files appear to only inlcude the filename and file data itself, which leads to empty creation date and permissions (bottom, screenshot 1). If I save the file itself without archiving it, it will contain some basic information (screenshot 2)

code: 

```swift
var entries: [TarEntry] = []
  for (_, imageData) in images.enumerated() {
      let filename = "\(self.lastIndex).jpeg"
      let tarEntry = TarContainer.Entry(info: .init(name: filename, type: .regular), data: imageData)
      entries.append(tarEntry)
      self.lastIndex+=1
    }
    do {
        let gzData = try GzipArchive.archive(data: TarContainer.create(from: entries))
        try gzData.write(to: archiveURL)
  
        print("Archive saved to \(archiveURL.path)")
    } catch {
        print("Error writing tar.gz data: \(error)")
    }
}
```

<img width="265" alt="image" src="https://github.com/tsolomko/SWCompression/assets/23126999/1005ef90-b68b-4dc5-98df-273dc6dd69a8">
<img width="249" alt="image" src="https://github.com/tsolomko/SWCompression/assets/23126999/25253413-54aa-41f3-a387-e808d5cd109a">

sorry if something is unclear, I'm normally a frontend developer so I don't really have any experience with swift whatsoever. :)

**Comment by tsolomko** [comment, marked as answer]
Hi,

The creation date, permissions and other info about files are empty, because when you create a TAR entry for each of the files you don't supply this information, and SWCompression does not make any assumptions about the default values of these properties (partially because a lot of them are optional, at least in the TAR case). Depending on your particular needs you may want to change the body of your for-loop to something like this

```swift
let filename = "\(self.lastIndex).jpeg"
var tarEntry = TarContainer.Entry(info: .init(name: filename, type: .regular), data: imageData)
tarEntry.info.permissions = Permissions(rawValue: 420) // This is an example (rw-r--r--), you may want to copy the permissions of the original file.
// ... Set other properties in tarEntry.info ...
entries.append(tarEntry)
self.lastIndex+=1
```

The list of available properties in `TarEntryInfo` can be found [here](https://www.tsolomko.me/SWCompression/Structs/TarEntryInfo.html).

**Reply by nick-delirium** [reply]
oh I see, thanks a lot, turns out the solution was under my nose the entire time :)

btw, thanks for your library. Its by far the best solution out of everything I've tried and the only one that actually lets you to create tar.gz in memory, which is crucial for my usecase. ❤️
